dbt/adapters/databricks/connections.py

```diff
@@ -797,6 +865,168 @@ def set_connection_name(
 
         return conn
 
+    # override
+    def release(self) -> None:
+        if not USE_LONG_SESSIONS:
+            return super().release()
+
+        with self.lock:
+            conn = cast(Optional[DatabricksDBTConnection], self.get_if_exists())
+            if conn is None:
+                return
+
+        conn._release()
+
+    # override
+    def cleanup_all(self) -> None:
+        if not USE_LONG_SESSIONS:
+            return super().cleanup_all()
+
+        with self.lock:
+            for thread_connections in self.threads_compute_connections.values():
+                for connection in thread_connections.values():
+                    if connection.acquire_release_count > 0 and connection.state not in {"init"}:
+                        fire_event(
+                            ConnectionLeftOpenInCleanup(conn_name=cast_to_str(connection.name))
+                        )
+                    else:
+                        fire_event(
+                            ConnectionClosedInCleanup(conn_name=cast_to_str(connection.name))
+                        )
+                    self.close(connection)
+
+            # garbage collect these connections
+            self.thread_connections.clear()
+            self.threads_compute_connections.clear()
+
+    def _get_compute_connection(
+        self, name: Optional[str] = None, node: Optional[ResultNode] = None
+    ) -> Connection:
+        """Called by 'set_connection_name' in DatabricksConnectionManager.
+        Creates a connection for this thread/node if one doesn't already
+        exist, and will rename an existing connection."""
+
+        _long_sessions_only("_get_compute_connection")
+
+        conn_name: str = "master" if name is None else name
+
+        # Get a connection for this thread
+        conn = self._get_if_exists_compute_connection(_get_compute_name(node) or "")
+
+        if conn is None:
+            conn = self._create_compute_connection(conn_name, node)
+        else:  # existing connection either wasn't open or didn't have the right name
+            conn = self._update_compute_connection(conn, conn_name, node)
+
+        conn._acquire(node)
+
+        return conn
+
+    def _update_compute_connection(
+        self,
+        conn: DatabricksDBTConnection,
+        new_name: str,
+        node: Optional[ResultNode] = None,
+    ) -> DatabricksDBTConnection:
+        """Update a connection that is being re-used with a new name, handle, etc."""
+        _long_sessions_only("_update_compute_connection")
+
+        compute_name = _get_compute_name(node=node) or ""
+        if conn.name == new_name and conn.state == "open" and conn.compute_name == compute_name:
```

Comment:
apologies if this is enforced somewhere I'm not seeing in the PR, but are we enforcing this to lower?  We've hit issues in the past where we compare to a particular case and the DBR decides to change what case they send.

Reply:
I copied this from BaseConnectionManager, but you make a good point.  I will make it more robust.